x-pack/test/monitoring_api_integration/README.md

```diff
@@ -0,0 +1,53 @@
+data are defined under `x-pack/test/api_integration/apis/monitoring`._
+
+Since an elastic-agent integration spawns the corresponding metricbeat module under
+the hood (ie when an agent policy defines elasticsearch metrics data streams,
+a metricbeat process with the elasticsearch module will be spawned), the output
+documents are _almost_ identical. This means that we can easily transform documents
+from a source (elastic-agent) to another (metricbeat), and have the same tests run
+against both datasets.
+
+Note that we don't have to install anything for the metricbeat data since the mappings
+are already installed by elasticseach at startup, and available at `.monitoring-<component>-8-mb`
+patterns. So we are always running the metricbeat tests against the latest version of
+the mappings.
+We could have a similar approach for packages, for example by installing the latest
+packages versions from public EPR before the test suites run, instead of using pinned
+versions. Besides the questionable reliance on remote services for running tests,
+this is also dangerous given that packages are released in a continuous model.
+This means that whenever the test suite would execute against the latest version
+of packages it would be too late, as in already available to users.
+
+### Validating a new package version
+- Get the locally built package from `<integrations-repo>/build/packages/<package>-<version>.zip`; or
+- Download the package zip at `<package-registry-url>/epr/<package>/<package>-<version>.zip`
+- Add the zipped package of the new version under `./fixtures/packages` and remove the previous version
+- Update the package version in `./packages.ts`
+- Create draft PR with the change to run against CI; or
+- Run the impacted test suite locally:
+  - start test server: `node scripts/functional_tests_server --config x-pack/test/monitoring_api_integration/config`
+  - start test suite for the updated component (eg Elasticsearch, Kibana..): `node scripts/functional_test_runner --config x-pack/test/monitoring_api_integration/config --grep "<Component>"`
+
+### Adding a new archive
+- Generate elastic-agent data for the relevant integration and archive the data
+  with the kbn-es-archiver. Assuming the data is extracted from an elastic-package
+  stack: `node scripts/es_archiver.js save <output-dir> 'metrics-<component>.stack_monitoring.*' --es-url=https://elastic:changeme@localhost:9200 --es-ca=~/.elastic-package/profiles/default/certs/ca-cert.pem`
+  - `<output-dir>` should point to a subdirectory of the `./archives` dir for consistency,
+    and since we're generating package data, end with `package` dir. example `<kibana>/x-pack/test/monitoring_api_integration/archives/kibana/two-nodes/package`
+- `<output-dir>` will contain a `data.json.gz` and a `mappings.json` file. Remove the `mappings.json`
+- run the transform script to generate metricbeat data `node scripts/transform_archive --src <output-dir>/data.json.gz`
```

Comment:
Just to see if I understood this:
`es_archiver` generated file will be stored in `./archives/package` and a file created by the new `transform_archive` script will be stored in `metricbeat` folder?

Reply:
Correct. the change include [examples](https://github.com/elastic/kibana/tree/72113db132aa780806d4a53314d5a93488384528/x-pack/test/monitoring_api_integration/archives/apm)